.github/workflows/release_droid_upload_github_release_assets.yml

```diff
@@ -8,8 +8,126 @@ on:
+          devtools::build(binary=TRUE, path='${{ env.output_dir }}')
+        shell: Rscript {0}
+        env: # Set individual GH PAT
+          GITHUB_PAT: ${{ secrets.PAT_FOR_DEVTOOLS }}
+      - name: Check result
+        run: |
+          echo 'Target folder is ${{ env.output_dir }}'
+          ls '${{ env.output_dir }}'
+      - name: Check result
+        if: matrix.config.os == 'macOS-latest'
+        run: |
+          pushd '${{ env.output_dir }}'
+          ls .
+          for f in *.tgz; do mv "$f" "$(echo "$f" | sed s/\.tgz/-${{matrix.config.os}}\.tgz/)"; done
+          ls .
+        shell: bash
+      - name: Check result
+        if: matrix.config.os == 'windows-latest'
+        run: |
+          pushd '${{ env.output_dir }}'
+          ls .
+          for f in *.zip; do mv "$f" "$(echo "$f" | sed s/\.zip/-${{matrix.config.os}}\.zip/)"; done
+          ls .
+        shell: bash
+      - name: Upload assets to the GitHub release draft
+        uses: shogo82148/actions-upload-release-asset@v1
+        with:
+          upload_url: ${{ github.event.inputs.upload_url }}
+          asset_path: ${{ runner.temp }}/bin/${{matrix.config.os}}/*
+  build-source:
     runs-on: ubuntu-latest
+    environment: ci_build
     steps:
-      - name: Dummy step
-        run: echo "Nothing to do"
+      - uses: actions/checkout@v2
+      - uses: r-lib/actions/setup-r@v2
+        with:
+          r-version: 'release'
+      - name: Install package dependencies
+        run: |
+          sudo apt-get -y update && \
+             sudo apt-get -y install --no-install-recommends libcurl4-openssl-dev curl libxml2-dev  libssl-dev && \
+             sudo locale-gen en_US.UTF-8 && \
+             sudo update-locale LC_ALL=en_US.UTF-8 && \
+             sudo apt-get -y clean && \
+             sudo apt-get -y autoremove
+      - name: Query r-dependencies
+        run: |
+          install.packages("RODBC")
+          install.packages("DBI")
+          install.packages("devtools", dependencies=TRUE)
+          devtools::install_deps()
+        shell: Rscript {0}
+        env: # Set individual GH PAT
+          GITHUB_PAT: ${{ secrets.PAT_FOR_DEVTOOLS }}
+      - name: Create output dir
+        run: |
+          out_put_dir="${{ runner.temp }}/source"
+          mkdir -p "${out_put_dir}"
+          echo "output_dir=${out_put_dir}" >> $GITHUB_ENV
+      - name: Build
+        run: |
+          devtools::build(binary=FALSE, path="${{ env.output_dir }}")
+        shell: Rscript {0}
+        env: # Set individual GH PAT
+          GITHUB_PAT: ${{ secrets.PAT_FOR_DEVTOOLS }}
+      - name: Check result
+        run: |
+           echo "Target folder is ${{ env.output_dir }}"
+           ls "${{ env.output_dir }}"
+      - name: Upload assets to the GitHub release draft
+        uses: shogo82148/actions-upload-release-asset@v1
+        with:
+          upload_url: ${{ github.event.inputs.upload_url }}
+          asset_path: ${{ runner.temp }}/bin/source/*
```

Comment:
We also try to upload sha256 sums of artifacts, maybe good idea also to upload them. Example from java: https://github.com/exasol/hive-virtual-schema/blob/main/.github/workflows/release_droid_upload_github_release_assets.yml#L32-L43

Reply:
This of course can be done in another ticket :)

Reply:
I think that's a great idea.
Although the objective is to provide packages to be used with the devtools::install_url() function, and I believe this function will not check the sha file, it's good to have it if anybody writes it's own install scripts.

I will add this right now to this PR.